src/request.ts

```diff
@@ -80,6 +80,7 @@ async function httpRequest<T = any>(
     const response = await fetch(url, {
         method,
         headers: {
+            "Content-Type": method === "POST" || method === "PATCH" ? "application/json" : "*/*",
```

Comment:
`Content-Type` only makes sense when there is a payload body. I don't think it would break anything to send `"*/*"` on a GET, but we ought to avoid it. 

Also, inserting POST/PUT handing inside the generic request function defeats some of our separation of concerns. The `post` and `patch` functions above provide a better place to do that. For example:
```
export function post<T = any>(
    service: SalesforceService,
    path: string,
    body?: Record<string, any>,
    headers?: Record<string, any>,
): Promise<T> {
    const postHeaders = {
        ...(body ? { "Content-Type": "application/json" } : undefined),
        ...headers,
    };
    return httpRequest(service, "POST", path, undefined, body, postHeaders);
}
```

Reply:
fixed